contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql

```diff
@@ -3754,3 +3754,154 @@ BEGIN
 END;
 $$ LANGUAGE pltsql;
 GRANT EXECUTE ON PROCEDURE sys.sp_procedure_params_100_managed TO PUBLIC;
+
+CREATE OR REPLACE VIEW sys.all_database_users AS
+SELECT DISTINCT
+CASE 
+  WHEN Ext.orig_username = 'dbo' THEN Base3.oid
+  WHEN Ext.orig_username = 'guest' THEN 0
+  ELSE Base2.oid
+END AS oid
+FROM pg_catalog.pg_roles AS Base INNER JOIN sys.babelfish_authid_user_ext AS Ext
+ON Base.rolname = Ext.rolname
+LEFT OUTER JOIN pg_catalog.pg_roles Base2
+ON Ext.login_name = Base2.rolname
+LEFT OUTER JOIN sys.babelfish_sysdatabases AS Db
+ON Ext.database_name COLLATE sys.database_default = Db.name
+LEFT OUTER JOIN pg_catalog.pg_roles AS Base3
+ON Db.owner = Base3.rolname
+WHERE Ext.type != 'R' AND Ext.orig_username IS NOT NULL;
+GRANT SELECT on sys.all_database_users TO PUBLIC;
+
+CREATE OR REPLACE VIEW sys.db_role_mapping AS
+SELECT
+	UExt2.database_name as database_name,
+	UExt1.orig_username as role_name,
+	UExt2.login_name as member_login
+FROM pg_catalog.pg_auth_members AS Authmbr
+INNER JOIN pg_catalog.pg_roles AS PGR1 ON PGR1.oid = Authmbr.roleid
+INNER JOIN pg_catalog.pg_roles AS PGR2 ON PGR2.oid = Authmbr.member
+INNER JOIN sys.babelfish_authid_user_ext AS UExt1 ON PGR1.rolname = UExt1.rolname
+INNER JOIN sys.babelfish_authid_user_ext AS UExt2 ON PGR2.rolname = UExt2.rolname
+WHERE UExt1.orig_username IN ('db_securityadmin', 'db_accessadmin');
+GRANT SELECT on sys.db_role_mapping TO PUBLIC;
+
+CREATE OR REPLACE PROCEDURE sys.sp_helplogins_internal_logins() 
+LANGUAGE pltsql
+AS $$
+BEGIN
+	IF is_srvrolemember('securityadmin') = 0 
+  BEGIN
+    RAISERROR('User does not have permission to perform this action.', 16, 1);
+		RETURN 1;
+  END
+
+	SELECT
+    CAST(LExt.orig_loginname AS sys.SYSNAME) AS LoginName,
+    CAST(CAST(Base.oid AS INT) AS sys.varbinary(85)) AS sid,
+    CAST(LExt.default_database_name AS SYS.SYSNAME) AS DefDBName,
+    CAST(LExt.default_language_name AS SYS.SYSNAME) AS DefLangName,
+    CASE 
+      WHEN Dp.oid IS NOT NULL THEN 'YES'
+      ELSE 'NO'
+    END as AUser,
+    'NO' AS ARemote -- Currently we do not support linking local logins to remote logins
+  FROM pg_catalog.pg_roles AS Base 
+  INNER JOIN sys.babelfish_authid_login_ext AS LExt ON Base.rolname = LExt.rolname
+  LEFT JOIN sys.all_database_users Dp ON Dp.oid = Base.oid -- In order to find out if a login has any users associated with it
```

Comment:
Instead for defining a separate view, please expand `all_database_users` here itself.